### PR TITLE
Improve tests coverage in db-config-store

### PR DESCRIPTION
**Pull request by norascheuch**
## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Comment by charisk** [comment]
Can we also add some tests for `setSelectedDbItem` please?

**Comment by norascheuch** [comment]
@charisk I added tests for all methods that check item existence as well!

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts [hunk 1]
```diff
@@ -184,6 +185,146 @@ describe("db config store", () => {
     });
   });
 
+  describe("add db items", () => {
```
Comment (charisk) [inline extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:188]: We could follow the same pattern in naming as the other tests e.g. 
```suggestion
  describe("db and list addition", () => {
```

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts [hunk 2]
```diff
@@ -184,6 +185,146 @@ describe("db config store", () => {
...
+      });
+
+      configPath = join(tempWorkspaceStoragePath, "workspace-databases.json");
+    });
+
+    it("should add a remote repository", async () => {
+      // Initial set up
+      const dbConfig = createDbConfig();
+
+      await writeJSON(configPath, dbConfig);
```
Comment (charisk) [inline extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:205]: (Optional) We could extract this and the next 2 lines into a function and re-use across all the tests.

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts [hunk 3]
```diff
@@ -496,4 +637,180 @@ describe("db config store", () => {
...
+      const updatedDbConfig = (await readJSON(configPath)) as DbConfig;
+
+      // Check that the config file has been updated
+      expect(updatedDbConfig.selected).toEqual(selectedItem);
+
+      configStore.dispose();
+    });
+  });
+
+  describe("does exist methods", () => {
```
Comment (charisk) [inline extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:677]: (Optional) Perhaps

```suggestion
  describe("existence checks", () => {
```

**Review by charisk — APPROVED** [review]
LGTM! Thanks for improving the test coverage.

Just some optional suggestions.

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts (hunk 3 above)
Comment (norascheuch) [reply extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:677]: 👍

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts (hunk 1 above)
Comment (norascheuch) [reply extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:188]: 👍

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts (hunk 2 above)
Comment (norascheuch) [reply extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:205]: Good idea! Done!

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts [hunk 4]
```diff
@@ -496,4 +601,178 @@ describe("db config store", () => {
...
+  });
+
+  async function initializeConfig(
+    dbConfig: DbConfig,
+    configPath: string,
+    app: App,
+  ): Promise<DbConfigStore> {
+    await writeJSON(configPath, dbConfig);
+
+    const configStore = new DbConfigStore(app);
```
Comment (shati-patel) [inline extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:773]:
```suggestion
    const configStore = new DbConfigStore(app, false);
```

There's now an option to initialize the config store without starting the file watcher, which is what we generally want in tests. Let's do that here too!

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts [hunk 5]
```diff
@@ -496,4 +601,178 @@ describe("db config store", () => {
...
+        extensionPath,
+        workspaceStoragePath: tempWorkspaceStoragePath,
+      });
+
+      configPath = join(tempWorkspaceStoragePath, "workspace-databases.json");
+    });
+
+    it("should set the selected item", async () => {
+      // Initial set up
+      const configStore = new DbConfigStore(app);
```
Comment (shati-patel) [inline extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts:620]:
```suggestion
      const configStore = new DbConfigStore(app, false);
```

As above!

**Review by shati-patel — APPROVED** [review]
Quick comment with suggestions from #1951!

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts (hunk 4 above)
Comment (norascheuch) [reply extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts]: 👍

File: extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts (hunk 5 above)
Comment (norascheuch) [reply extensions/ql-vscode/test/unit-tests/databases/config/db-config-store.test.ts]: 👍

**Pull request merged**